src/EditorFeatures/Core/EditorConfigSettings/Data/Whitespace/WhitespaceSetting.cs

```diff
@@ -56,5 +69,79 @@ protected WhitespaceSetting(string description, OptionUpdater updater, SettingLo
             var location = new SettingLocation(isDefinedInEditorConfig ? LocationKind.EditorConfig : LocationKind.VisualStudio, fileName);
             return new WhitespaceSetting<TOption>(option, description, editorConfigOptions, visualStudioOptions, updater, location);
         }
+
+        private IEditorConfigStorageLocation2? GetEditorConfigStorageLocation()
+        {
+            return Key.Option.StorageLocations.OfType<IEditorConfigStorageLocation2>().FirstOrDefault();
+
+        }
+
+        public string? GetSettingName()
+        {
+            var storageLocation = GetEditorConfigStorageLocation();
+            return storageLocation?.KeyName;
+        }
+
+        public string GetDocumentation()
+        {
+            return Description;
+        }
+
+        public ImmutableArray<string>? GetSettingValues(OptionSet optionSet)
+        {
+            var storageLocation = GetEditorConfigStorageLocation();
```

Comment:
Can you add a todo here to switch this over to the editorconfig value holder (with issue tracking it?)